client/llb/definition.go

```diff
@@ -156,3 +159,27 @@ func (d *DefinitionOp) platform() *specs.Platform {
 	d.platforms[d.dgst] = platform
 	return platform
 }
+
+func StatesFromDefinitions(defs map[string]*pb.Definition) (map[string]State, error) {
+	states := make(map[string]State)
+	for key, def := range defs {
+		defop, err := NewDefinitionOp(def)
+		if err != nil {
+			return nil, err
+		}
+		states[key] = NewState(defop)
+	}
+	return states, nil
+}
+
+func DefinitionsFromStates(states map[string]State) (map[string]*pb.Definition, error) {
```

Comment:
These seem unnecessarily defined as public methods in `llb` package to me. As all they do is call an already public method with a slice input. One of them is called only once, other twice so I think this logic can be on the caller side.

Reply:
I'm happy to move them to the caller side, but there's actually 2 + 3 calls and they're in different packages.
```
client/solve.go:193:20:		defs, err := llb.DefinitionsFromStates(opt.FrontendInputs)
frontend/gateway/gateway.go:735:19:	defs, err := llb.DefinitionsFromStates(lbf.inputs)
frontend/gateway/grpcclient/client.go:320:19:		defs, err = llb.DefinitionsFromStates(creq.FrontendInputs)
control/control.go:277:21:	inputs, err := llb.StatesFromDefinitions(req.FrontendInputs)
frontend/gateway/gateway.go:458:21:	inputs, err := llb.StatesFromDefinitions(req.FrontendInputs)
```

Reply:
I think if we change the `frontend.Frontend` interface to take `pb.Definition` we can reduce the number of calls and then I'll just move these to the caller side.

Reply:
After changing it, it reduced to one call each so I moved them to the caller side.